src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -2985,12 +2985,12 @@ CalcASHRAESimpExtConvectCoeff(DataSurfaces::SurfaceRoughness const Roughness, //
     Real64 CalcASHRAESimpExtConvectCoeff;
 
     // FUNCTION PARAMETER DEFINITIONS:
-    static Array1D<Real64> const D({0, 5}, {11.58, 12.49, 10.79, 8.23, 10.22, 8.23});
-    static Array1D<Real64> const E({0, 5}, {5.894, 4.065, 4.192, 4.00, 3.100, 3.33});
-    static Array1D<Real64> const F({0, 5}, {0.0, 0.028, 0.0, -0.057, 0.0, -0.036});
+    constexpr std::array<Real64, 6> const D = {11.58, 12.49, 10.79, 8.23, 10.22, 8.23};
+    constexpr std::array<Real64, 6> const E = {5.894, 4.065, 4.192, 4.00, 3.100, 3.33};
+    constexpr std::array<Real64, 6> const F = {0.0, 0.028, 0.0, -0.057, 0.0, -0.036};
```

Comment:
Global `constexpr std::array`s essentially live in the read-only-data segment and are loaded from the binary. Do local `constexpr std::array`s also live there? It seems like it should easy for the compiler to figure out that they can and should instead of recreating them in the stack frame? Do you need to add `static` to tell the compiler to move these arrays out of the stack frame? It probably doesn't hurt, but you shouldn't need to do it and ... a quick test program shows that you in fact don't need to.

Reply:
That sounds good, I was basing my uncertainty on this answer which concluded that it could be helpful to leave static on there: https://stackoverflow.com/a/13867690

Reply:
Admittedly, the example I wrote was very simple (the array was in `main` which is called only once and I based my conclusion by looking at the assembly implementation which accessed the array using `%rip` which is the instruction pointer in the text segment). It is certainly simpler than the example in the second answer on that page (the top answer is not very good and I am surprised it got so many votes). So maybe in more complicated situations it does help to add `static`. It certainly doesn't hurt to add it, so sure go ahead.